app/views/vouchers/_redeem_editor_voucher.html.erb

```diff
@@ -0,0 +1,24 @@
+<p data-cy="redeem-voucher-text">
+  <%= t('profile.become_editor_html', lecture: voucher.lecture.title) %>
+</p>
+
+<% if current_user == voucher.lecture.teacher %>
+
+  <p>
+    <%= t('profile.teacher_cant_become_editor') %>
+    <%= cancel_voucher_button %>
```

Comment:
If you put a redeem_voucher_button for this case, the teacher will remain the teacher but the editor will be removed from the lecture. That means, a teacher could possibly construct a POST request to do exactly this. But then, they could also do this directly in the UI by revoking the editor for the lecture, so we shouldn't bother. Just to let you know.

Reply:
Thanks for noticing!